adal/tests/Test.ADAL.NET.Unit/AdalDotNetTests.cs

```diff
@@ -698,7 +698,7 @@ public void ClientCredentialExtendedExpiryNoFlagSetTest()
             });
 
             //whitelisted authority
-            context = new AuthenticationContext(TestConstants.DefaultAuthorityCommonTenant, true);
+            context = new AuthenticationContext(TestConstants.DefaultAuthorityCommonTenant, true, new TokenCache());
```

Comment:
>, new TokenCache() [](start = 96, length = 18)

why is this needed?

Reply:
Without this, it would use the default static token cache, therefore this test case would not be stateless, it would leave tokens in the cache. The result would be each test cases can be run separately, but when we Run-All, some of them would fail.

Reply:
Copied from an earlier answer from github:

Without this, it would use the default static token cache, therefore this test case would not be stateless, it would leave tokens in the cache. The result would be each test cases can be run separately, but when we Run-All, some of them would fail.

---
In reply to: [205345693](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r205345693) [](ancestors = 205345693)